integration-tests/commons/src/main/java/io/quarkiverse/fury/it/FuryResources.java

```diff
@@ -16,8 +17,11 @@
 import org.apache.fury.serializer.Serializer;
 import org.apache.fury.util.Preconditions;
 
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
 @Path("/fury")
 @ApplicationScoped
+@RegisterForReflection
```

Comment:
@gastaldi I have to add `@RegisterForReflection` here to make `FuryResource` working in native mode. It looks like we need it when the JAX-RS reources are from the dependencies?

Reply:
Probably because it was removed from ArC? Tried with `@Unremovable`?

Reply:
The error message in native tests looks like 
```
2024-11-15 14:02:44,802 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /fury/test failed, error id: 9d9ff831-cab9-46ad-ab65-87ff019e74e4-1: java.lang.RuntimeException: java.lang.NoSuchMethodException: io.quarkiverse.fury.it.FuryResources.testBar(io.quarkiverse.fury.it.Bar)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:84)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getParameterAnnotations(ResteasyReactiveResourceInfo.java:125)
	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.getAnnotations(RequestDeserializeHandler.java:133)
	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.isReadable(RequestDeserializeHandler.java:118)
	at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:78)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:135)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:627)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@21.0.5/java.lang.Thread.runWith(Thread.java:1596)
	at java.base@21.0.5/java.lang.Thread.run(Thread.java:1583)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
Caused by: java.lang.NoSuchMethodException: io.quarkiverse.fury.it.FuryResources.testBar(io.quarkiverse.fury.it.Bar)
	at java.base@21.0.5/java.lang.Class.checkMethod(DynamicHub.java:1078)
	at java.base@21.0.5/java.lang.Class.getMethod(DynamicHub.java:1063)
	at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:79)
	... 18 more


```

I suppose it miss the reflection information. Will try `@Unremovable`

Reply:
@gastaldi `@Unremovable` does not work. Do you think this is an issue of Quarkus? I can prepare a simple reproducer.

Reply:
Any thoughts here @geoand ? 

@zhfeng a reproducer would help, sure

Reply:
I can look into it when there is a reproducer

Reply:
Well, it only fails to find the method below, the other metheds all look good. It's weird and any hint?
```java
    @POST
    @Path("/test")
    @Produces("application/fury")
    @Consumes("application/fury")
    public Bar testBar(Bar obj) {
        ...
   }
```

Reply:
Weird, could it be because it starts with `test`?

Reply:
The other fact is this test works with `quarkus-resteasy` which is classic rest. But fails with `quarkus-rest` which is reactive rest.

Reply:
OK, I have a reproducer with `quarkus-fury` and `quarkus-rest`. But I need to do a new temporary release of  `quarkus-fury` to include this PR.